software/32.Zig/README.md

```diff
@@ -0,0 +1,267 @@
+📚 Documentation:
+
+> 💡 Now, that you have created a file `main.zig`, you can use other zig files. To use them in your `main.zig` you have to integrate the module ([read more](https://stackoverflow.com/questions/71186556/how-do-i-include-one-zig-file-from-another-zig-file))
+
+- [Build System](https://ziglang.org/learn/build-system/)
+- [Doc.Zig](https://ziglang.org/documentation/master/)
+
+✔️ Validation:
+
+    you should see the following :
+
+    ```sh
+    Hello, World!
+    ```
+
+## Step 2 - Palindrome?
+
+📑 Description:
+
+For the second exercise, you have to create a function that takes as parameter a string `word`.
+
+📌 Tasks:
+
+Create a file `palindrome.zig` for this new function.
+
+This function must return true if the word given in parameter is a palindrome and false in the opposite case.
+
+📚 Documentation:
+
+ -[What is a palindrome ?](https://www.wikiwand.com/en/Palindrome)
+ -[New function](https://ziglang.org/documentation/master/#Functions)
+ -[The types in Zig](https://ziglang.org/documentation/master/#Primitive-Types)
+ -[Control_Structures](https://zig.guide/language-basics/while-loops)
+
+ ✔️ Validation:
+
+    Here's a main example :
+
+    pub fn main() void {
+    const stdout = std.io.getStdOut().writer();
+
+    const test_word1 = "madam";
+    const test_word2 = "hello";
+
+    const result1 = is_palindrome(test_word1);
+    const result2 = is_palindrome(test_word2);
+
+    stdout.print("{} is palindrome: {}\n", .{test_word1, result1}) catch {};
+    stdout.print("{} is palindrome: {}\n", .{test_word2, result2}) catch {};
+}
+
+ When you compile and run palindrome.zig, the output should be:
+
+    ```sh
+    madam is palindrome: true
+    ```
+    or
+
+    ```sh
+    hello is palindrome: false
+    ```
+
+## Step 3 - Fibonacci sequence
+
+📑 Description:
+
+For the third exercise, you need to create a function that generates and displays the Fibonacci sequence up to a specified number of elements.
+
+📌 Tasks:
+
+- Create a file `fibonacci.zig` for this new function.
+
+For the third exercise, create a function that takes one parameter:
+
+- A number `max` with type `i32` that represent the number of element to compute.
+
+- You must now display the sequence of Fibonacci from the number of starts to the `max` value.
+
+Here is a small example of the beginning of the Fibonacci sequence:
+
+    ```shell
+    0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, ...
+    ```
+
+📚 Documentation:
+
+ -[Fibonacci number](https://www.wikiwand.com/en/Fibonacci_number)
+ -[Match](https://ziglang.org/documentation/master/#Standard-Library-Math-Functions)
+ -[Vector](https://ziglang.org/documentation/master/#Vectors)
+
+  ✔️ Validation:
+
+    Given the Fibonacci sequence (see previous example), if max is 5, the function should output:
+
+    ```sh
+    0, 1, 1, 2, 3
+    ```
+
+## Step 4 - A simple and useful `Calculator`
+
+📑 Description:
+
+Let's go further now!
+
+The objective of this fifth exercise is to create a simplifying calculator.
```

Comment:
Okey, I like the exercice, but you should give more details, like a graphical calculator, a interactive calculator in the terminal? You need to be the most precise possible.